### PR TITLE
pnpm@9.9

### PR DESCRIPTION
**Pull request by andresgnlez**
## Substitute this line for a meaningful title for your changes

Docker (client):
- makes sure dependencies are installed correctly by installing `shared` dependencies first and then `client` ones.
- removed redundant files copy.

Miscellaneous:
- Updated `@ts-rest` dependencies as part of the debugging process.
- Bumped pnpm to `9.9`
- Uses pnpm's catalog to centralize `zod` version.

### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

### CODE REVIEW

**Comment by andresgnlez** [comment]
@alexeh @alepefe-vizz please, test Docker locally and make sure there's nothing weird. I've updated the PR description with the changes made.

File: client/Dockerfile [hunk 1]
```diff
@@ -12,21 +12,16 @@ ENV AUTH_CREDENTIALS $AUTH_CREDENTIALS
...
-COPY client/package.json client/tsconfig.json ./client/
-COPY shared/package.json shared/tsconfig.json ./shared/
-
-RUN corepack enable pnpm
-
 COPY client ./client
 COPY shared ./shared
 
+RUN corepack enable pnpm
+RUN pnpm shared:deps
```
Comment (alexeh) [inline client/Dockerfile:20]: I understand the fix and why now it works, but it feels weird as fuck to see that pnpm does not handle this dependency graph on its own (when apparently in 9.4.x did)

Let's have a quick chat about it later cc @alepefe-vizz

**Review by alexeh — APPROVED** [review]

**Review by alexeh — APPROVED** [review]
Tested locally, it works like a charm. Thanks for taking care of this bro

File: client/Dockerfile (hunk 1 above)
Comment (andresgnlez) [reply client/Dockerfile]: We are resolving this by running `pnpm install` and letting `pnpm` handle the dependencies to install.

**Pull request merged**